doc/cprover-manual/jbmc-user-manual.md

```diff
@@ -0,0 +1,237 @@
+\ingroup module_hidden
+\page jbmc-user-manual JBMC User Manual
+
+
+\section jbmc-manual-intro Introduction
+
+JBMC is a bounded model checking tool for the Java language. It follows the
+same principles as CBMC, thus the existing
+[documentation for CBMC](http://www.cprover.org/cprover-manual/cbmc.html)
+largely applies to JBMC.
+
+In brief, JBMC creates a model from Java Bytecode (given as input as a .class
+file or a .jar) and performs static analysis and symbolic simulation which allow
+to prove or refute properties. These properties are either automatically
+generated, or provided by the user in the form of assertions.
+
+Using examples, the following section will guide you through the basic
+features of JBMC, and explore the most useful features.
+
+
+\section jbmc-tutorial JBMC tutorial
+
+\subsection jbmc-manual-auto-assert Automatic assertions
```

Comment:
Jbmc can detect two things: A. thrown exceptions/errors or B. uncaught exceptions/errors.

Orthogonally, there are 3 groups of errors that we want to detect:
1. Assertion violations 
2. Runtime exceptions 
3. Explicit throws

The question is how best to introduce it. B2, B3 are probably the concepts Java developers are most familiar with. A is a bit more for experts and probably best explained for 2 (as you do). 
For people familiar with assertions starting with A1 would be doable. The difference between A1 and B1 is for practical purposes irrelevant (unless you catch AssertionError somewhere).

In the long term I'd like to make B the default behaviour.

Reply:
What do you want me to do here? Shall I enumerate these groups in the introduction?

Reply:
We have to restructure the sections a bit to start with familiar concepts and then go to less familiar ones. But we can do this in a second iteration.